src/wcl/tracing.cpp

```diff
@@ -116,6 +116,40 @@ void error(std::initializer_list<std::pair<const std::string, std::string>> list
   log_message(LOG_LEVEL_ERROR, list, fmt, args);
 }
 
+void fatal(const char* fmt, ...) {
+  va_list args;
+  va_start(args, fmt);
+  auto defer = make_defer([&]() { va_end(args); });
+  log_message(LOG_LEVEL_FATAL, {}, fmt, args);
+  ::exit(1);
+}
+
+void fatal(std::initializer_list<std::pair<const std::string, std::string>> list, const char* fmt,
+           ...) {
+  va_list args;
+  va_start(args, fmt);
+  auto defer = make_defer([&]() { va_end(args); });
+  log_message(LOG_LEVEL_FATAL, list, fmt, args);
```

Comment:
I mean for `fatal` to call `error` and for `exit` to call `info` but then each of them do their respective exits

Reply:
We can keep the same LOG_LEVEL if you want, but we can't forward the va_args to error/info so they have to be complete functions instead of helpers

Reply:
honestly its already made it through CI, thats I would just land it if I were you.